cmd/osbuild-jobsite-builder/main.go

```diff
@@ -181,58 +172,42 @@ func (builder *Builder) HandleProvision(w http.ResponseWriter, r *http.Request)
 	_, err = io.Copy(dst, r.Body)
 
 	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
 		return fmt.Errorf("Builder.HandleProvision: Failed to write manifest.json")
 	}
 
-	w.WriteHeader(http.StatusCreated)
-
-	if _, err := w.Write([]byte(`done`)); err != nil {
-		return fmt.Errorf("Builder.HandleProvision: Failed to write response")
+	if err = dst.Close(); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return fmt.Errorf("Builder.HandleProvision: Failed to close manifest.json")
```

Comment:
Here (and above) discarding the original error means we close information, I think this should be something like: `		return fmt.Errorf("Builder.HandleProvision: Failed to close manifest.json: %v", err)`

Reply:
Agreed, I've changed all error conversions to include the original `err`.